### PR TITLE
Convert Buffer and node crypto to web apis

### PR DESCRIPTION
**Pull request by bitofbreeze**
This PR converts all uses of Buffer and node crypto APIs to Uint8array and web crypto.

This would be a major release because it bumps the minimum node engine version and converts all exports to be async.

This makes this package fully compatibility with serverless, and would make remix-auth-totp too https://github.com/dev-xo/remix-auth-totp/pull/74

If this isn't desired, I can make a new package.

### CODE REVIEW

**Review by kentcdodds — APPROVED** [review]
Thanks for this. I'm a fan. Bummer it makes this async and will require a major version bump, but it's fine.

File: index.js [hunk 1]
```diff
@@ -11,7 +9,7 @@ import base32Decode from 'base32-decode'
 // That said, if you're acting the role of both client and server and your TOTP
 // is longer lived, you can definitely use a more secure algorithm like SHA256.
 // Learn more: https://www.rfc-editor.org/rfc/rfc4226#page-25 (B.1. SHA-1 Status)
-const DEFAULT_ALGORITHM = 'SHA1'
+const DEFAULT_ALGORITHM = 'SHA-1'
```
Comment (kentcdodds) [inline index.js:12]: Can you help me understand why it was important to change these algorithm names? OTP clients break when you include the hyphen:

image

cc @dev-xo have you seen issues with this change?

Comment (kentcdodds) [reply index.js:12]: For now I'm just planning on doing this:

```tsx
const otpUri = getTOTPAuthUri({
	...verification,
	// OTP clients break with the `-` in the algorithm name.
	algorithm: verification.algorithm.replaceAll('-', ''),
	accountName: user.email,
	issuer,
})
```

But the other thing is this change necessitates a data migration if people saved the algorithm in the database and that kinda stinks. Would rather revert this change if it's not actually important.

Comment (bitofbreeze) [reply index.js:12]: web crypto apis use this spelling

Comment (dev-xo) [reply index.js:12]: No idea @kentcdodds.

I think the same changes were applied for `remix-auth-totp` if I'm not wrong, also the Strategy does not store the algorithm in the database, so probably those refactors were a bit safer for us.

Feel free to take the changes you think `epixweb-totp` may require Kent, as those probably do not affect us.
- - -
A question I have for @bitofbreeze, is `SHA1` and `SHA-1` _(hyphen-minus)_ the same, or is the algorithm recognized by writing it both ways?

Comment (bitofbreeze) [reply index.js:12]: Here are the possible values https://developer.mozilla.org/en-US/docs/Web/API/HmacImportParams

**Pull request merged**